### ISSUE TITLE
math.simplify reorders operations

### ISSUE THREAD

**Issue by LightWarrior** [josdejong/mathjs#933]
```javascript
math.simplify('x + y + a', [{l: 'n+-c', r: 'n-c'}], {a: -1}).toString()
```
gives
```javascript
"x - 1 + y"
```
The order of operations gets changed. Is it an issue?
Is there any way to just simplify signs (`'+-n' => '-n'`, `'--n' => 'n'`) without node tree getting rearranged so drastically (and get `x + y - 1` in my example)?

**Comment by firepick1** [comment]
I have no idea why the legacy simplify does what you saw, but I added your sensible simplification to simplifyCore since it can be done in a single pass. I added a new test that replicated your output and fixed it as suggested. I think the optimization is benign since the precedence of binary + and - are the same--but I'm only 98.1% sure. 😄  

https://github.com/josdejong/mathjs/commit/4787237b315572dd356e89eb78e62376179faac8

**Comment by LightWarrior** [comment]
Nice! Thank you!

**Comment by firepick1** [comment]
@Jos will determine next steps.

**Comment by josdejong** [comment]
This should be fixed now in `v3.16.3`, thanks @firepick1 !
